### PR TITLE
71x validate memory

### PR DESCRIPTION
**Pull request by JumaX**
# Description

This PR adds additional hosts checks to confirm that their is enough memory to install and run the given component.  Note, this is specifically to run the component based on its default heap size with extra buffer (1GB) for the operating system.  It is not considered to be the recommended setting for a production environment.

Fixes # (issue)

ANSIENG-256

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

This has been validated by running the mtls-customcerts-rhel scenario and confirm that all checks pass in addition to the replicator memory check being skipped as it's not available in this scenario.  This confirms that logic is correct for when to run the given check.

`TASK [confluent.platform.common : Validate enough free memory for Zookeeper hosts] ***
skipping: [kafka-broker1]
skipping: [kafka-broker2]
skipping: [kafka-broker3]
skipping: [schema-registry1]
skipping: [kafka-connect1]
skipping: [ksql1]
skipping: [control-center1]
skipping: [kafka-rest1]
ok: [zookeeper1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [confluent.platform.common : Validate enough free memory for Kafka Broker hosts] ***
skipping: [zookeeper1]
skipping: [schema-registry1]
skipping: [kafka-connect1]
skipping: [ksql1]
skipping: [control-center1]
skipping: [kafka-rest1]
ok: [kafka-broker1] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [kafka-broker2] => {
    "changed": false,
    "msg": "All assertions passed"
}
ok: [kafka-broker3] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [confluent.platform.common : Validate enough free memory for Kafka Connect hosts] ***
skipping: [zookeeper1]
skipping: [kafka-broker1]
skipping: [kafka-broker2]
skipping: [kafka-broker3]
skipping: [schema-registry1]
skipping: [ksql1]
skipping: [control-center1]
skipping: [kafka-rest1]
ok: [kafka-connect1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [confluent.platform.common : Validate enough free memory for Schema Registry hosts] ***
skipping: [zookeeper1]
skipping: [kafka-broker1]
skipping: [kafka-broker2]
skipping: [kafka-broker3]
skipping: [kafka-connect1]
skipping: [ksql1]
skipping: [control-center1]
skipping: [kafka-rest1]
ok: [schema-registry1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [confluent.platform.common : Validate enough free memory for KSQL hosts] ***
skipping: [zookeeper1]
skipping: [kafka-broker1]
skipping: [kafka-broker2]
skipping: [kafka-broker3]
skipping: [schema-registry1]
skipping: [kafka-connect1]
skipping: [control-center1]
skipping: [kafka-rest1]
ok: [ksql1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [confluent.platform.common : Validate enough free memory for Rest Proxy hosts] ***
skipping: [zookeeper1]
skipping: [kafka-broker1]
skipping: [kafka-broker2]
skipping: [kafka-broker3]
skipping: [schema-registry1]
skipping: [kafka-connect1]
skipping: [ksql1]
skipping: [control-center1]
ok: [kafka-rest1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [confluent.platform.common : Validate enough free memory for Control Center hosts] ***
skipping: [zookeeper1]
skipping: [kafka-broker1]
skipping: [kafka-broker2]
skipping: [kafka-broker3]
skipping: [schema-registry1]
skipping: [kafka-connect1]
skipping: [ksql1]
skipping: [kafka-rest1]
ok: [control-center1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [confluent.platform.common : Validate enough free memory for Kafka Connect Replicator hosts] ***
skipping: [zookeeper1]
skipping: [kafka-broker1]
skipping: [kafka-broker2]
skipping: [kafka-broker3]
skipping: [schema-registry1]
skipping: [kafka-connect1]
skipping: [ksql1]
skipping: [control-center1]
skipping: [kafka-rest1]`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/common/tasks/host_validations.yml [hunk 1]
```diff
@@ -78,14 +78,110 @@
...
       to continue installation. To skip host validations, set validate_hosts to false."
   with_items: "{{ ansible_mounts }}"
-  tags: 
+  tags:
     - validate
     - validate_disk_usage
+
+- name: Validate enough free memory for Zookeeper hosts
+  assert:
+    that: ansible_memtotal_mb > {{ required_total_memory_mb_zookeeper }}|float
```
Comment (nsharma-git) [inline roles/common/tasks/host_validations.yml:95]: Does it not make sense to use * ansible_memfree_mb*

File: roles/common/tasks/host_validations.yml [hunk 2]
```diff
@@ -78,14 +78,110 @@
...
+    fail_msg: >-
+      "Not enough memory to run Zookeeper.  Minimum memory required is {{ required_total_memory_mb_zookeeper }}MB, please add more memory to host
+      or choose a different host to install Zookeeper to continue installation."
+  when: groups['zookeeper'] is defined and inventory_hostname in groups['zookeeper']
+  tags:
+    - validate
+    - validate_memory_usage
+    - validate_memory_usage_zookeeper
+
+- name: Validate enough free memory for Kafka Broker hosts
```
Comment (nsharma-git) [inline roles/common/tasks/host_validations.yml:105]: Can we use some loop here to reuse the code

Comment (JumaX) [reply roles/common/tasks/host_validations.yml:105]: Can you clarify where you think a loop should be specified here?

File: roles/common/tasks/host_validations.yml (hunk 1 above)
Comment (JumaX) [reply roles/common/tasks/host_validations.yml:95]: Based on our internal discussions it does not.  As when discussed with product, this is to be treated as an installer to make sure that basic hardware requirements are met, and not about the utilisation of said hardware.

Comment (JumaX) [reply roles/common/tasks/host_validations.yml:95]: Thinking about this further, I think you are right.  Let me update this.

File: roles/common/tasks/host_validations.yml (hunk 2 above)
Comment (nsharma-git) [reply roles/common/tasks/host_validations.yml:134]: Nevermind, not much of benefit.

**Review by nsharma-git — APPROVED** [review]

**Pull request merged**